src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionDiagnosticAnalyzer.cs

```diff
@@ -28,30 +28,89 @@ protected override void InitializeWorker(AnalysisContext context)
             {
-                Exception: IConversionOperation
+                if (operation is IThrowOperation
+                    {
+                        Exception: IConversionOperation
+                        {
+                            Operand: IObjectCreationOperation
+                            {
+                                Constructor.ContainingType: INamedTypeSymbol constructedType,
+                            },
+                        },
+                        Syntax: var throwSyntax
+                    } throwOperation &&
+                    notImplementedExceptionType.Equals(constructedType))
                 {
-                    Operand: IObjectCreationOperation
+                    // Report diagnostic for each throw operation
+                    context.ReportDiagnostic(Diagnostic.Create(
+                        Descriptor,
+                        throwOperation.Syntax.GetLocation()));
+
+                    // If the throw is the top-level operation in the containing symbol, report a diagnostic on the
+                    // symbol as well. Note: consider reporting on the entire symbol, instead of just the name.  And in
+                    // this case, do not report directly on the throw as well.
+                    if (IsTopLevel(block, operation))
                     {
-                        Constructor.ContainingType: INamedTypeSymbol constructedType,
-                    },
-                },
-                Syntax: ThrowExpressionSyntax or ThrowStatementSyntax,
-            } &&
-            notImplementedExceptionType.Equals(constructedType))
+                        foreach (var location in context.OwningSymbol.Locations)
+                        {
+                            if (location.SourceTree == context.FilterTree)
+                            {
+                                context.ReportDiagnostic(Diagnostic.Create(
+                                    Descriptor,
+                                    location));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private static bool IsTopLevel(IOperation block, IOperation operation)
+    {
+        if (block is IBlockOperation { Operations: [var child] })
         {
-            context.ReportDiagnostic(Diagnostic.Create(
-                Descriptor,
-                throwOperation.Syntax.GetLocation()));
+            // Handle: { throw new NotImplementedException(); }
+            if (child == operation)
+                return true;
+
+            // Handle: => throw new NotImplementedException();
+            if (child is IReturnOperation { ReturnedValue: IConversionOperation { Operand: var operand } } && operand == operation)
+            {
+                if (operation.Syntax is ThrowExpressionSyntax throwExpression)
+                {
+                    // Exclude property declarations with expression bodies
```

Comment:
why?

Reply:
because they would otherwise doubly report

Reply:
renamed method and added comments so the code should now have more clarity